conditions/operator_conditions.go

```diff
@@ -119,15 +125,21 @@ func IsConditionStatusPresentAndEqual(operatorCondition *api.OperatorCondition,
 	return false, nil
 }
 
-// getOperatorNamespace returns the namespace the operator should be running in.
+// getOperatorNamespace returns the namespace the operator should be running in. It
+// reads the serviceaccount namespace file if present to find the namespace. If the file
+// is not present, namespace can be passed through env variable "OPERATOR_NAMESPACE_LOCAL"
 func getOperatorNamespace() (string, error) {
-	nsBytes, err := readNamespace()
+	ns, err := utils.GetOperatorNamespace(readNamespace)
 	if err != nil {
-		if os.IsNotExist(err) {
-			return "", errors.New("cannot find namespace of the operator")
+		errMsg := "cannot find namespace of the operator"
+		if errors.Is(err, utils.ErrNoNamespace) {
+			ns = os.Getenv(operatorNsEnvVar)
+			if ns == "" {
+				return "", fmt.Errorf(" %s: environment variable %s not set", errMsg, operatorCondEnvVar)
+			}
+			return ns, nil
 		}
-		return "", err
+		return "", fmt.Errorf("%s: %v", errMsg, err)
 	}
-	ns := strings.TrimSpace(string(nsBytes))
 	return ns, nil
```

Comment:
Based on my comment above, remove this function and call `utils.GetOperatorNamespace()` directly?

Reply:
This would mean that `getOperatorNamespace` is present in both the packages, instead of the utils package. Declaring the error twice and checking for the error from utils.GetOperatorNamespace() to be equal to to the one defined in `leader.go` would not work.